### PR TITLE
single delete

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
@siddontang @BusyJay @ngaut PTAL

### CODE REVIEW

File: src/rocksdb.rs [hunk 1]
```diff
@@ -1284,6 +1338,41 @@ mod test {
             assert!(name.to_str().unwrap().find("LOG").is_none());
         }
     }
+
+    #[test]
+    fn single_delete_test() {
+        let path = TempDir::new("_rust_rocksdb_singledeletetest").expect("");
+        let db = DB::open_default(path.path().to_str().unwrap()).unwrap();
+
+        db.put(b"a", b"v1").unwrap();
```
Comment (siddontang) [inline src/rocksdb.rs]: How about `single_delete` the nonexistent key?

Comment (zhangjinpeng1987) [inline src/rocksdb.rs]: Similar to Delete() method, SingleDelete() removes the database entry for a key, but has the prerequisites that the key exists and was not overwritten.

Comment (siddontang) [inline src/rocksdb.rs]: so `single_delete` a nonexistent key is undefined?

Comment (zhangjinpeng1987) [inline src/rocksdb.rs]: yes

**Comment by zhangjinpeng1987** [comment]
PING

**Comment by siddontang** [comment]
LGTM

**Comment by zhangjinpeng1987** [comment]
@BusyJay @disksing  PTAL

**Comment by BusyJay** [comment]
LGTM

**Comment by disksing** [comment]
LGTM

**Pull request merged**